IHP/Controller/Param.hs

```diff
@@ -203,6 +204,13 @@ instance ParamReader Double where
             Right value -> Right value
             Left error -> Left ("ParamReader Dobule: " <> cs error)
 
+instance ParamReader Float where
+    {-# INLINE readParameter #-}
+    readParameter byteString =
+        case Attoparsec.parseOnly (Attoparsec.double <* Attoparsec.endOfInput) byteString of
+            Right value -> Right (Float.double2Float value)
```

Comment:
Are there situations where this conversion could catch people out?

Reply:
I've been thinking about this too. It could happen that a user types in a number with lots of precision which can only be represented as a double. In that case a bit of the precision would be lost during conversion. Practically I don't think will cause problems, as usually when you have lots of precision as a requirement you will use double anyway.

Attoparsec sadly doesn't provide a float parser, and we also cannot write the parser ourselves as the required helper function for that is not exported by the attoparsec module.